src/Adapter/Image/ImageGenerator.php

```diff
@@ -68,13 +68,13 @@ public function __construct(FeatureFlagRepository $featureFlagRepository, ImageF
      * @throws ImageOptimizationException
      * @throws ImageUploadException
      */
-    public function generateImagesByTypes(string $imagePath, array $imageTypes): bool
+    public function generateImagesByTypes(string $imagePath, array $imageTypes, int $imageId): bool
```

Comment:
This is a BC break: if you add a mandatory argument to a PHP function, users that do use the previous signature will have a PHP fatal error.

There is 2 options to avoid the BC break (forbidden in minor versions)
1. Add optional default value
2. Create a new function and if eligible deprecate the previous one

Reply:
Thanks @matks for the clarification :smiley: 

In both cases, if the user uses the function outside the current context, it won't work and there will be side effects